internal/status/svc_annotation.go

```diff
@@ -70,18 +70,25 @@ func DeleteSvcAnnotation(key, namespace, name string) {
 	if _, found := ann[lib.NPLSvcAnnotation]; !found {
 		return
 	}
-	delete(ann, lib.NPLSvcAnnotation)
-	patchPayload := map[string]interface{}{
-		"metadata": map[string]map[string]string{
-			"annotations": ann,
+
+	payloadValue := make(map[string]*string)
+	for key, val := range ann {
+		payloadValue[key] = &val
+	}
```

Comment:
Any reason you are setting back the old annotations to the service? For deleting a specific annotation, this is not required, the statement below will do, as this is not a `replace` operation but a `merge` operation.

Reply:
Thanks ! I thought it would replace the old annotations. I'll remove it.